### PR TITLE
fix genderef of jsgen

### PR DESCRIPTION
**Pull request by hsssnow23**
fix "invalid left-hand side in assignment" in js backend.

about error case, please look tderef test.

### CODE REVIEW

File: compiler/jsgen.nim [hunk 1]
```diff
@@ -1200,7 +1200,8 @@ proc genDeref(p: PProc, n: PNode, r: var TCompRes) =
       r.res = "$1[$2]" % [a.address, a.res]
     elif n.sons[0].kind == nkCall:
       let tmp = p.getTemp
-      r.res = "($1 = $2, $1[0][$1[1]])" % [tmp, a.res]
+      addf(p.body, "$1 = $2;$n", [tmp, a.res])
```
Comment (Araq) [inline compiler/jsgen.nim]: I think this would break some invariant in the codegen. @yglukhov ?

**Comment by yglukhov** [comment]
The only issue I can see is that it may change argument evaluation order. But maybe its not that critical...
Another issue i found while testing this (though not caused by this PR) is #5275.

**Comment by hsssnow23** [comment]
How about this commit?

**Comment by yglukhov** [comment]
👍 cool!

**Pull request merged**